python_modules/dagster/dagster/_core/execution/context/asset_execution_context.py

```diff
@@ -457,17 +457,26 @@ def add_output_metadata(
         metadata: Mapping[str, Any],
         output_name: Optional[str] = None,
         mapping_key: Optional[str] = None,
+        partition_key: Optional[str] = None,
     ) -> None:
-        return self.op_execution_context.add_output_metadata(
-            metadata=metadata, output_name=output_name, mapping_key=mapping_key
+        return self._step_execution_context.add_output_metadata(
+            metadata=metadata,
+            output_name=output_name,
+            mapping_key=mapping_key,
+            asset_partition_key=partition_key,
```

Comment:
why partition_key above and asset_partition_key here?

I think just using `partition_key` everywhere makes sense

Reply:
I kept it as asset_partition_key here to make it clear that this is only really applicable to asset events, and not to partitioned ops for example

Reply:
Yea I'm fine with that. Since it doesn't make it into the underlying output object